### DISCUSSION TITLE
ND-LotTenderingProcess node in group GR-Lot-Purpose (notice types not unified criteria)

### DISCUSSION THREAD

**Discussion by AlexAS81** [OP-TED/eForms-SDK#573] [Q&A] [answered]
In the notice types from 1 to 6 (1.json, 2.json, 3.json, 4.json, 5.json, 6.json) in the group "GR-Lot-Purpose" it is not indicated the "ND-LotTenderingProcess" nodeId, in the rest of the notice types there is.

Group in 4.json:
![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/ebeaff9f-bcc7-4b94-9025-ee905ba6092c)

Group in 10.json:
![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/e3b758a4-eb51-4aaf-aee7-1bbc35548edd)

If we look at the fields.json we understand that the correct thing is that this group (GR-Lot-Purpose) does not take the nodeId (ND-LotTenderingProcess).

This makes it impossible for us to correctly generate the conceptual model. We understand that the notice types should be unified and the same groups should have the same nodeId. Is there an intention to correct this? Or do we have to adapt the SDK to make it work? The lack of unification of criteria is making it quite difficult for us to use the SDK.

Thanks in advance
Alex

**Comment by jaggaer-fgeniola** [comment, marked as answer]
Hi Alex,
I had the same problem and had to implement a recursive root search to find the correct parent node.
Unfortunately without this information in the SDK I think we have no other way.
And it's not even the only lack in the SDK, we hope in the next updates because basing all our work on incomplete data is not a sustainable choice.
Regards
